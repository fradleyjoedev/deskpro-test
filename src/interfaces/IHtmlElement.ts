export interface IHTMLElement {
  componentType: string;
  title: string;
  description: string;
}