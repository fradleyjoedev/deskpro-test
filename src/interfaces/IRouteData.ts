export interface IRouteData {
  path: string;
  component: string;
  linkTitle: string;
  pageTitle: string;
  pageDescription: string;
}