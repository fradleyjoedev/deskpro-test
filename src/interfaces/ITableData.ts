export interface ITableData {
  heading: string;
  data: string[];
};