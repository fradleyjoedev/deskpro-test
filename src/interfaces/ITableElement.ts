export interface ITableElement {
  componentType: string;
}