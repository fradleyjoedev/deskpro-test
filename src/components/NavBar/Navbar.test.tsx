import React from "react";
import { shallow } from 'enzyme';
import { NavBar } from './index';
import { ITableElement } from '../../interfaces/ITableElement';
import { IRouteData } from "##/interfaces/IRouteData";

const data: IRouteData[] = [{
  path: "/page-a",
  items: [],
  linkTitle: "Page A",
  pageTitle: "Info about Page A"
}]

describe('Test for NavBar', () => {
  it('snapshot', () => {
    const link = shallow(<NavBar links={data} />);
    expect(link).toMatchSnapshot();
  });
});