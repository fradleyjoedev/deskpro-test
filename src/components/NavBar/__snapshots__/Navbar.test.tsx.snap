// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NavBar snapshot 1`] = `
<ul
  className="navbar"
>
  <Link
    key="0"
    onClick={[Function]}
    to="/page-a"
  >
    <li
      className="navbar__item"
    >
      Page A
    </li>
  </Link>
</ul>
`;
